### ISSUE TITLE
Bug: statusline goes blank when I reload init.lua

### ISSUE THREAD

**Issue by aegatlin** [hoob3rt/lualine.nvim#276]
### Self Checks

- [x ] I'm using the latest lualine and neovim nightly.
- [x ] I didn't find the issue in exsisting issues or prs.

This is potentially related to #230, but that issue thread seems focused on statusline never showing (or quickly blinking away), while mine is about _going_ blank on config file reload. Before a config reload, lualine works exactly as desired.

Also, in #230, it sounded like @hoob3rt was talking about a plugin reload issue in [this comment](https://github.com/hoob3rt/lualine.nvim/issues/230#issuecomment-841431134), which seems like it's related to this issue I'm creating now, but I'm confused on what the resolution is. And, finally, since my situation is easily reproducible, it felt like creating a new issue was a good idea. (Sorry if I'm missing something.)

### How to reproduce the problem

Open vim and run `:luafile $MYVIMRC` (or a keymapped equivalent)

### Actual behaviour

The statusline goes blank.

### Expected behaviour

The statusline is maintained.

### Additional information

`:echo $MYVIMRC` => `/Users/[username]/.config/nvim/init.lua` (on MacOS).

In my init.lua is `require 'lualine'.setup()`. I suspect the error is related to repeat calls to `setup()`. 

I am constantly playing with mapping and so constantly reloading my init.lua file. I'd say it's part of my daily workflow. There does not seem to be a problem with reloading init.lua with any other plugins I'm using.

This is a great plugin. Thank you for making it :D

**Comment by shadowninja55** [comment]
same here, it goes blank after a reload. it also goes blank when i run something with `:!` if that helps.

**Comment by jhchabran** [comment]
I faced the same issue while working on a theme with multiple variants that can be changed at the runtime. 
A workaround I found is to unload `lualine` entirely before calling `setup(...)` again. 

```lua
  require("plenary.reload").reload_module("lualine", true)
  require("lualine").setup({...})
```

**Comment by shadowninja55** [comment]
> I faced the same issue while working on a theme with multiple variants that can be changed at the runtime.
> A workaround I found is to unload `lualine` entirely before calling `setup(...)` again.
> 
> ```lua
>   require("plenary.reload").reload_module("lualine", true)
>   require("lualine").setup({...})
> ```

ooh that worked thanks

**Comment by aegatlin** [comment]
Well, seems like we got a good solution here, thanks to @jhchabran , but I still reckon this should stay open until there's an internal fix for this. For my setup, it's the only plugin where I'm doing this.

**Comment by jhchabran** [comment]
I strongly agree with you @aegatlin, unloading everything to reload is really radical. 

The simplest way to reproduce the bug is to call `setup` twice in a row, meaning it's probably caused by some state issues. Ideally, the setup should be able to detect that it has been called already and if that's the case just update the configuration.

**Comment by Hubro** [comment]
Just to chime in, lualine is currently the only plugin I use (out of almost 30) that breaks when I reload my configuration, so this is definitely not the norm :smile: Fortunately, reloading the package with plenary does work, thanks for the tip @jhchabran. Just unloading the package first with `packages.loaded.lualine = nil` didn't work, so I was close to giving up and installing something else.

**Comment by jhchabran** [comment]
I'm glad it could help!

> Just unloading the package first with `packages.loaded.lualine = nil` didn't work, so I was close to giving up and installing something else.

@Hubro Did you unload all modules this way? `lualine.*` I think that's what prevented it to work. That's basically what the `true` flag does in the above snippet.

**Comment by shadmansaleh** [comment]
Resolved in #311

**Comment by aegatlin** [comment]
Problem has been resolved for me. Thank you!
